sdk/core/Azure.Core/src/Shared/TaskExtensions.cs

```diff
@@ -15,41 +19,43 @@ public static T EnsureCompleted<T>(this Task<T> task)
 #if DEBUG
             VerifyTaskCompleted(task.IsCompleted);
 #endif
-#pragma warning disable AZC0102
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
             return task.GetAwaiter().GetResult();
-#pragma warning restore AZC0102
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
         }
 
         public static void EnsureCompleted(this Task task)
         {
 #if DEBUG
             VerifyTaskCompleted(task.IsCompleted);
 #endif
-#pragma warning disable AZC0102
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
             task.GetAwaiter().GetResult();
-#pragma warning restore AZC0102
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
         }
 
         public static T EnsureCompleted<T>(this ValueTask<T> task)
         {
 #if DEBUG
             VerifyTaskCompleted(task.IsCompleted);
 #endif
-#pragma warning disable AZC0102
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
             return task.GetAwaiter().GetResult();
-#pragma warning restore AZC0102
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
         }
 
         public static void EnsureCompleted(this ValueTask task)
         {
 #if DEBUG
             VerifyTaskCompleted(task.IsCompleted);
 #endif
-#pragma warning disable AZC0102
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
             task.GetAwaiter().GetResult();
-#pragma warning restore AZC0102
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult(). Use the TaskExtensions.EnsureCompleted() extension method instead.
         }
 
+        public static Enumerable<T> EnsureSyncEnumerable<T>(this IAsyncEnumerable<T> asyncEnumerable) => new Enumerable<T>(asyncEnumerable);
```

Comment:
Should this be called `EnsureMoveNextCompleted` or similar to help make it clear that we're checking the Task completed?  I'd love some variant of `Ensure(.*)Completed` so the rare occasions we'll see this will be easier to associate with the more common `EnsureCompleted`.

Reply:
@pakrym , do you have any opinion on this?